### PR TITLE
Di 26648 need to migrate all import relation to have init relative url

### PR DESCRIPTION
**Pull request by OdedAnswer**

### CODE REVIEW

File: server-side/services/collections.service.ts [hunk 1]
```diff
@@ -187,21 +187,21 @@ export class CollectionsService {
         // only check for field's type when there are Fields on the collection
         if (collectionObj.Fields) {
             for (const fieldID of Object.keys(collectionObj.Fields!)) {
-                if(collectionObj.Fields![fieldID].Type === 'ContainedResource'){ // if field type is contained, count contained schema fields
+                if (collectionObj.Fields![fieldID].Type === 'ContainedResource') { // if field type is contained, count contained schema fields
                     const collectionFieldsCount = await this.getCollectionFieldsLength(collectionObj.Fields![fieldID].Resource!);
```
Comment (michaelatias) [inline server-side/services/collections.service.ts:191]: why await inside for of?

File: server-side/installation.ts [hunk 2]
```diff
@@ -27,17 +28,29 @@ export async function uninstall(client: Client, request: Request): Promise<any>
 
 export async function upgrade(client: Client, request: Request): Promise<any> {
     
-    // if we are upgrading from a version before 0.9.20, create a relation to var settings and settings table
+    // if we are upgrading from a version before 0.9.20, create a relation to var settings and settings table and also migrate DIMX relations
     if (request.body.FromVersion && semver.compare(request.body.FromVersion, '0.9.20') < 0) {
-        return await createObjects(client);
+        await createObjects(client);
+        return await migrateDIMXRelations(client);
```
[the remark this answers is not in the document]
Comment (michaelatias) [reply server-side/installation.ts:34]: you can do something like that
let resultObject = {success:true,resultObject:{}}
if from older version than 0.9.20:
    create objects
if from older version than 0.9.27 
    resultObject = migrate ()
 return resultObject

File: server-side/services/collections.service.ts [hunk 3]
```diff
@@ -329,29 +329,44 @@ export class CollectionsService {
...
             for (const collection of collections) {
-                if (collection.Type != 'contained') {
-                    // if the collection has no indexed fields, we need to delete the relation.
-                    if (!this.globalService.isCollectionIndexed(collection)) {
-                        // we're making a hack by setting the collection as Hidden, the relation will be Hidden as well
-                        collection.Hidden = true;
-                    }
-                    await this.createDataQueryRelations(collection);
-                }
+                await this.createDIMXRelations(collection);
```
Comment (michaelatias) [inline server-side/services/collections.service.ts:336]: dont use for of with await

File: server-side/services/collections.service.ts [hunk 4]
```diff
@@ -329,29 +329,44 @@ export class CollectionsService {
...
+        }
+    }
+
+    async unmigrateDIMXRelations() {
+        try {
+            const relations = await this.utilities.papiClient.addons.data.relations.find({ where: `RelationName='DataImportResource' and AddonUUID='${this.client.AddonUUID}'` });
+            // for each relation we need to set the InitRelationDataRelativeURL to undefined
+            for (const relation of relations) {
+                relation.InitRelationDataRelativeURL = '';
+                await this.utilities.papiClient.addons.data.relations.upsert(relation);
```
Comment (michaelatias) [inline server-side/services/collections.service.ts:358]: same here for with await

File: server-side/services/collections.service.ts (hunk 1 above)
Comment (OdedAnswer) [reply server-side/services/collections.service.ts:191]: was this changed? I didn't write this

File: server-side/services/collections.service.ts (hunk 4 above)
Comment (michaelatias) [reply server-side/services/collections.service.ts:354]: you need to return the promises inside the map

**Pull request merged**